packages/clerk-js/src/ui/elements/FormControl.tsx

```diff
@@ -53,6 +64,55 @@ const getInputElementForType = (type: FormControlProps['type']) => {
   return CustomInputs[customInput] || Input;
 };
 
+function useFormTextAnimation() {
+  const prefersReducedMotion = usePrefersReducedMotion();
+
+  const getFormTextAnimation = useCallback(
+    (enterAnimation: boolean): ThemableCssProp => {
+      if (prefersReducedMotion) {
+        return {
+          animation: 'none',
+        };
+      }
+      return t => ({
+        animation: `${enterAnimation ? animations.inAnimation : animations.outAnimation} 300ms ${
+          t.transitionTiming.$common
+        }`,
+      });
+    },
+    [prefersReducedMotion],
+  );
+
+  return {
+    getFormTextAnimation,
+  };
+}
+
+const useCalculateErrorTextHeight = () => {
+  const [height, setHeight] = useState(24);
+
+  const calculateHeight = useCallback((element: HTMLElement | null, messageToDisplay: string | undefined) => {
```

Comment:
Can we simplify this and use the `offsetHeight` property. If we want to calculate something before it's in the dom, we can always add it with `position:absolute` and `opacity:0` and calculate the values, although I don't think we need that in this case since this receives an element.

Reply:
I gave this a try:
- When the callback runs you get access to the element, offsetHeight is 0. Interestingly if wait for 100ms inside a setTimeout the offsetHeight is now a proper value i can use. Sadly this delays the animation.

I am not quite sure why this is happening as the element.innerText seems to be the appropriate text value.

Reply:
Folks, if I get it right we want to animate height but the final height is still being determined due to the variable text. For such cases, there are three other CSS techniques that can be used:

1. Work with max-height and height: auto
2. Use scale-y
3. Use flexbox

Have we tried these? The max-height one might work for us we don't have very big variations between the hardcoded max-height value we will pick and the actual height.

For more information please refer to https://css-tricks.com/using-css-transitions-auto-dimensions/.

Reply:
I will give these a try, but the animation are already very efficient, the only inefficient part is that we change the height dynamically, this only happens in very rare cases and based on the most common flows a user this may not even get triggered.

Reply:
I don't see how the above methods will prevent the height only increasing and not decreasing in order to avoid the elements below bouncing when we are changing between neutral/error/success states.